### PR TITLE
FI-678: Update clinical notes sequence to check all patients

### PR DESCRIPTION
**Pull request by Jammjammjamm**
This branch updates the clinical notes guidance sequence to check all patient ids for each note type rather than only checking the first patient.

**Submitter:**
- [x] This pull request describes why these changes were made
- [x] Internal ticket links to this PR
- [x] Internal ticket is properly labeled (Community/Program)
- [x] Internal ticket has a justification for its Community/Program label
- [x] Code diff has been reviewed for extraneous/missing code
- n/a Tests are included and test edge cases
- [x] Tests/code quality metrics have been run locally and pass

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: lib/modules/us_core_guidance/clinicalnotes_sequence.rb [hunk 1]
```diff
@@ -52,57 +52,65 @@ class USCoreR4ClinicalNotesSequence < SequenceBase
...
 
-        reply = get_resource_by_params(versioned_resource_class(resource_class), search_params)
-        assert_response_ok(reply)
-        assert_bundle_response(reply)
+          reply = get_resource_by_params(versioned_resource_class(resource_class), search_params)
+          assert_response_ok(reply)
+          assert_bundle_response(reply)
 
-        resources_found = reply&.resource&.entry&.any? { |entry| entry&.resource&.resourceType == resource_class }
+          next if reply&.resource&.entry&.none? { |entry| entry&.resource&.resourceType == resource_class }
```
Comment (arscan) [inline lib/modules/us_core_guidance/clinicalnotes_sequence.rb]:
```suggestion
          next unless reply&.resource&.entry&.any? { |entry| entry&.resource&.resourceType == resource_class }
```

Comment (arscan) [reply lib/modules/us_core_guidance/clinicalnotes_sequence.rb]: I was just wondering what if entry evaluates to nil?  Probably should trigger that next?  That might not be a valid bundle, in which case this is moot, but all those safe operators made me wonder what would happen if nil is returned somewhere in the chain.  Do check my logic though if you agree (it is a little mind-bending).

File: lib/modules/us_core_guidance/clinicalnotes_sequence.rb [hunk 2]
```diff
@@ -52,57 +52,65 @@ class USCoreR4ClinicalNotesSequence < SequenceBase
...
+        resources_found = patient_ids.any? do |patient_id|
+          search_params = { 'patient': patient_id, 'category': category_code }
 
-        resources_found = reply&.resource&.entry&.any? { |entry| entry&.resource&.resourceType == resource_class }
+          reply = get_resource_by_params(versioned_resource_class(resource_class), search_params)
+          assert_response_ok(reply)
+          assert_bundle_response(reply)
 
-        skip "No #{resource_class} resources with category #{category_code} appear to be available. Please use patients with more information." unless resources_found
+          next if reply&.resource&.entry&.none? { |entry| entry&.resource&.resourceType == resource_class }
```
Comment (arscan) [inline lib/modules/us_core_guidance/clinicalnotes_sequence.rb]: same comment as above.

File: lib/modules/us_core_guidance/clinicalnotes_sequence.rb (hunk 1 above)
Comment (Jammjammjamm) [reply lib/modules/us_core_guidance/clinicalnotes_sequence.rb]: Yeah, good catch.
